src/avt/Database/Ghost/avtUnstructuredDomainBoundaries.C

```diff
@@ -1793,159 +1915,144 @@ avtUnstructuredDomainBoundaries::CommunicateMeshInformation(
-                T *fromPtr = (T *)(givingUg ->GetPoints()
-                                                    ->GetVoidPointer(0));
-
-                T *gainedPtr = gainedPtrStart;
-                int *origIdPtr = origIdPtrStart;
-
-                int k;
-                for (k = 0; k < nPts; ++k)
+                std::vector<std::array<T, 3>> gainedPtsToSend(nPts);
+                std::vector<int> origIdsToSend(nPts);
+                T *fromPtr = static_cast<T *>(givingUg->GetPoints()->GetVoidPointer(0));
+                for (int ptId = 0; ptId < nPts; ptId ++)
                 {
-                    *(origIdPtr++) = givenPoints[index][k];
+                    const int origPtId = givenPoints[index][ptId];
+                    origIdsToSend[ptId] = origPtId;
 
-                    T *ptPtr = fromPtr + 3 * givenPoints[index][k];
-                    *(gainedPtr++) = *(ptPtr++);
-                    *(gainedPtr++) = *(ptPtr++);
-                    *(gainedPtr++) = *(ptPtr++);
+                    T *ptPtr = fromPtr + 3 * origPtId;
+                    gainedPtsToSend[ptId][0] = *(ptPtr++);
+                    gainedPtsToSend[ptId][1] = *(ptPtr++);
+                    gainedPtsToSend[ptId][2] = *(ptPtr++);
                 }
 
                 // Send the number of points given
                 MPI_Send(&nPts, 1, MPI_INT, tRank, mpiNPtsTag, VISIT_MPI_COMM);
 
                 // Send the gained points
-                MPI_Send(gainedPtrStart, nPts * 3, type, tRank,
+                MPI_Send(gainedPtsToSend.data(), nPts * 3, type, tRank,
                          mpiGainedPointsTag, VISIT_MPI_COMM);
 
                 // Send the original ids for the gained points
-                MPI_Send(origIdPtrStart, nPts, MPI_INT, tRank,
+                MPI_Send(origIdsToSend.data(), nPts, MPI_INT, tRank,
                          mpiOriginalIdsTag, VISIT_MPI_COMM);
 
                 // Send the number of given cells
-                int nCells = (int)givenCells[index].size();
+                const int nCells = static_cast<int>(givenCells[index].size());
                 MPI_Send(&nCells, 1, MPI_INT, tRank, mpiNumGivenCellsTag,
                          VISIT_MPI_COMM);
 
                 // Prepare for sending the cell info
-                int *cellPtr = new int[nCells];
-                vector<int> cellPtsVector;
-                int *nPtsPerCellPtr = new int[nCells];
-
-                for (int j = 0; j < nCells; ++j)
+                std::vector<int> cellDataToSend(nCells);
+                std::vector<int> cellPtsToSend;
+                std::vector<int> nPtsPerCellToSend(nCells);
+                for (int cellId = 0; cellId < nCells; cellId ++)
                 {
-                    cellPtr[j] = givingUg->GetCellType(givenCells[index][j]);
-                    givingUg->GetCellPoints(givenCells[index][j], idList);
+                    cellDataToSend[cellId] = givingUg->GetCellType(givenCells[index][cellId]);
```

Comment:
I forgot to ask about this one....Is this array really storing VTK cell types? If so, should its type really be [`vtkCellType`](https://vtk.org/doc/nightly/html/vtkCellType_8h_source.html)

Reply:
I'm inclined to agree, but at the same time, we later send this with MPI as an integer array. There isn't an option for `vtkCellType` arrays in MPI. So at some point, we will treat it as an integer array, the question is where.

Reply:
Ok. Lets leave as is then.